tools/sense_studio/sense_studio.py

```diff
@@ -300,7 +302,79 @@ def add_class(project):
     for split in ['train', 'valid']:
         videos_dir = os.path.join(path, f'videos_{split}')
         class_dir = os.path.join(videos_dir, class_name)
-        os.mkdir(class_dir)
+
+        if not os.path.exists(class_dir):
+            os.mkdir(class_dir)
+
+    return redirect(url_for("project_details", path=path))
+
+
+@app.route('/edit-class/<string:project>/<string:class_name>', methods=['POST'])
+def edit_class(project, class_name):
+    """
+    Edit the class name and tags for an existing class in the given project.
+    """
+    project = urllib.parse.unquote(project)
+    class_name = urllib.parse.unquote(class_name)
+
+    # Lookup project path
+    projects = _load_project_overview_config()
+    path = projects[project]['path']
+
+    # Get new class name and tags
+    data = request.form
+    new_class_name = data['className']
+    new_tag1 = data['tag1'] or f'{new_class_name}_tag1'
+    new_tag2 = data['tag2'] or f'{new_class_name}_tag2'
+
+    if new_tag2 == new_tag1:
+        new_tag1 = f'{new_tag1}_1'
+        new_tag2 = f'{new_tag2}_2'
```

Comment:
Just a nit but do we want to dry this since it's almost the same as the block in `add_class`?

Reply:
Yes, thanks for calling me out on that :wink: